integration/test_web.py

```diff
@@ -0,0 +1,155 @@
+import sys
+import time
+import shutil
+import json
+import urllib2
+from os import mkdir, unlink, utime
+from os.path import join, exists, getmtime
+
+import util
+
+import requests
+import pytest_twisted
+
+
+def test_index(alice):
+    """
+    we can download the index file
+    """
+    util.web_get(alice._node_dir, "")
+    # ...and json mode is valid json
+    json.loads(util.web_get(alice._node_dir, "?t=json"))
+
+
+def test_upload_download(alice):
+    """
+    upload a file, then download it via readcap
+    """
+
+    FILE_CONTENTS = "some contents"
+
+    readcap = util.web_post(
+        alice._node_dir,
+        "uri",
+        data={
+            "t": "upload",
+            "format": "mdmf",
+        },
+        files={
+            "file": FILE_CONTENTS,
+        },
+    )
+    readcap = readcap.strip()
+    print("readcap '{}'".format(readcap))
+
+    data = util.web_get(
+        alice._node_dir, "uri",
+        params={
+            "uri": readcap,
+            "filename": "boom",
+        }
+    )
+    assert data == FILE_CONTENTS
+
+
+def test_put(alice):
+    """
+    use PUT to create a file
+    """
+
+    FILE_CONTENTS = "added via PUT"
+
+    resp = requests.put(
+        util.node_url(alice._node_dir, "uri"),
+        files={
+            "file": FILE_CONTENTS,
+        },
+    )
+    assert resp.text.strip().startswith("URI:CHK:")
+    assert resp.text.strip().endswith(":2:4:153")
```

Comment:
It would be good to parse this with ... allmydata.uri.from_string` and then make some more structured assertions.